compiler/src/dmd/expressionsem.d

```diff
@@ -875,6 +875,13 @@ Lagain:
                 if (sc.setUnsafePreview(global.params.systemVariables, false, loc,
                     "cannot access `@system` variable `%s` in @safe code", sd))
                 {
+                    if (auto v = sd.isVarDeclaration())
+                    {
+                        if (v.systemInferred)
+                            errorSupplemental(v.loc, "`%s` is inferred to be `@system` from its initializer here", v.toChars());
```

Comment:
The if-else statement can be replaced with a conditional expression in the call to errorSupplemental:

```
errorSupplemental(v.loc, v.systemInferred ? "`%s` is inferred to be `@system` from its initializer here" : "`%s` is declared here", v.toChars());
```

Reply:
I think it reads better with each error having its own statement

Reply:
expressions are cheaper, here the statement introduces a new layer of semantic analysis that is of no utility.